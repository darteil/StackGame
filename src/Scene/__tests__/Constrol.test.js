import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Controls from '../Control';

describe('Message component', () => {
  const onClick = jest.fn();
  const output = shallow(<Controls restartGame={onClick} />);

  it('should render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('test click event', () => {
    output.find('button').simulate('click');
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
