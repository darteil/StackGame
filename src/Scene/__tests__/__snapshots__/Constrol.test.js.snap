// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render correctly 1`] = `
<div>
  <button
    className=""
    onClick={[MockFunction]}
  >
    Restart
  </button>
</div>
`;
